File: app/src/main/java/co/aspirasoft/sams/ActivityDashboard.java
package co.aspirasoft.sams;

import android.os.Bundle;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBar;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.AdapterView;
import android.widget.LinearLayout;
import android.widget.ListView;

public class ActivityDashboard extends AppCompatActivity {

    //private String[] mMenuItems;
    private DrawerLayout mDrawerLayout;
    private ActionBarDrawerToggle mDrawerToggle;
    private LinearLayout mDrawerMain;
    private ListView mDrawerList;
    private ActionBar actionBar;

    private String mTitle;
    private String mDrawerTitle;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_dashboard);

        setSupportActionBar((Toolbar) findViewById(R.id.actionBar));
        actionBar = getSupportActionBar();
        actionBar.setHomeAsUpIndicator(R.drawable.ic_drawer);
        setTitle("ActivityDashboard");

        //mMenuItems = getResources().getStringArray(R.array.menu_items);
        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
        mDrawerMain = (LinearLayout) findViewById(R.id.left_drawer);
        mDrawerList = (ListView) findViewById(R.id.listView);

        mDrawerToggle = new ActionBarDrawerToggle(
                this,                                   /* host Activity */
                mDrawerLayout,                          /* DrawerLayout object */
                (Toolbar) findViewById(R.id.actionBar), /* nav bar */
                R.string.drawer_open,                   /* "open drawer" description */
                R.string.drawer_close                   /* "close drawer" description */
        ) {

            /** Called when a drawer has settled in a completely closed state. */
            public void onDrawerClosed(View view) {
                super.onDrawerClosed(view);
                setTitle(mTitle);
            }

            /** Called when a drawer has settled in a completely open state. */
            public void onDrawerOpened(View drawerView) {
                super.onDrawerOpened(drawerView);
                setTitle(mDrawerTitle);
            }
        };

        // Set the drawer toggle as the DrawerListener
        mDrawerLayout.setDrawerListener(mDrawerToggle);

        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setHomeButtonEnabled(true);

        // Set the adapter for the list view
        //final TypedArray typedArray = getResources().obtainTypedArray(R.array.menu_icons);
        //mDrawerList.setAdapter(new ArrayAdapter<String>(
        //        this,
        //        R.layout.drawer_items,
        //        mMenuItems
        //) {
            //@Override
            //public View getView(int position, View convertView, ViewGroup parent) {
                //View v = super.getView(position, convertView, parent);
                //int resourceId = typedArray.getResourceId(position, 0);
                //Drawable drawable = getResources().getDrawable(resourceId);
                //((TextView) v).setCompoundDrawablesWithIntrinsicBounds(drawable, null, null, null);
                //return v;
            //}
        //});

        // Set the list's click listener
        mDrawerList.setOnItemClickListener(new DrawerItemClickListener());
    }

    /**
     * Swaps views in the main content view
     */
    private void selectItem(int position) {
        // Hide all views
        findViewById(R.id.dashboardHome).setVisibility(View.INVISIBLE);
        findViewById(R.id.dashboardClasses).setVisibility(View.INVISIBLE);
        findViewById(R.id.dashboardAssignments).setVisibility(View.INVISIBLE);
        findViewById(R.id.dashboardPayments).setVisibility(View.INVISIBLE);
        findViewById(R.id.dashboardHostel).setVisibility(View.INVISIBLE);
        findViewById(R.id.dashboardSettings).setVisibility(View.INVISIBLE);
        findViewById(R.id.dashboardHelp).setVisibility(View.INVISIBLE);

        // Show appropriate view and update page title
        switch (position) {
            case 0:
                findViewById(R.id.dashboardHome).setVisibility(View.VISIBLE);
                setTitle("ActivityDashboard");
                break;
            case 1:
                findViewById(R.id.dashboardClasses).setVisibility(View.VISIBLE);
                //setTitle(mMenuItems[1]);
                break;
            case 2:
                findViewById(R.id.dashboardAssignments).setVisibility(View.VISIBLE);
                //setTitle(mMenuItems[2]);
                break;
            case 3:
                findViewById(R.id.dashboardPayments).setVisibility(View.VISIBLE);
                //setTitle(mMenuItems[3]);
                break;
            case 4:
                findViewById(R.id.dashboardHostel).setVisibility(View.VISIBLE);
                //setTitle(mMenuItems[4]);
                break;
            case 5:
                findViewById(R.id.dashboardSettings).setVisibility(View.VISIBLE);
                //setTitle(mMenuItems[5]);
                break;
            case 6:
                findViewById(R.id.dashboardHelp).setVisibility(View.VISIBLE);
                //setTitle(mMenuItems[6]);
                break;
        }

        // Highlight the selected item, update the title, and close the drawer
        mDrawerList.setItemChecked(position, true);
        mDrawerLayout.closeDrawer(mDrawerMain);
    }

    @Override
    public void setTitle(CharSequence title) {
        mTitle = mDrawerTitle = title.toString();
        actionBar.setTitle(title);
    }

    private class DrawerItemClickListener implements ListView.OnItemClickListener {
        @Override
        public void onItemClick(AdapterView parent, View view, int position, long id) {
            selectItem(position);
        }
    }
}

File: app/src/main/java/co/aspirasoft/sams/ui/activity/ActivityDashboard.kt
package co.aspirasoft.sams.ui.activity

import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.LinearLayout
import android.widget.ListView
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.drawerlayout.widget.DrawerLayout
import co.aspirasoft.sams.R

class ActivityDashboard : AppCompatActivity() {

    //private String[] mMenuItems;
    private lateinit var mDrawerLayout: DrawerLayout
    private lateinit var mDrawerToggle: ActionBarDrawerToggle
    private lateinit var mDrawerMain: LinearLayout
    private lateinit var mDrawerList: ListView
    private lateinit var mTitle: String
    private lateinit var mDrawerTitle: String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_dashboard)

        setSupportActionBar(findViewById<View>(R.id.actionBar) as Toolbar)
        supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_drawer)
        title = "ActivityDashboard"

        //mMenuItems = getResources().getStringArray(R.array.menu_items);
        mDrawerLayout = findViewById(R.id.drawer_layout)
        mDrawerMain = findViewById(R.id.left_drawer)
        mDrawerList = findViewById(R.id.listView)
        mDrawerToggle = object : ActionBarDrawerToggle(
            this,  /* host Activity */
            mDrawerLayout,  /* DrawerLayout object */
            findViewById<View>(R.id.actionBar) as Toolbar,  /* nav bar */
            R.string.drawer_open,  /* "open drawer" description */
            R.string.drawer_close /* "close drawer" description */
        ) {
            /**
             * Called when a drawer has settled in a completely closed state.
             */
            override fun onDrawerClosed(view: View) {
                super.onDrawerClosed(view)
                title = mTitle
            }

            /** Called when a drawer has settled in a completely open state.  */
            override fun onDrawerOpened(drawerView: View) {
                super.onDrawerOpened(drawerView)
                title = mDrawerTitle
            }
        }

        // Set the drawer toggle as the DrawerListener
        mDrawerLayout.setDrawerListener(mDrawerToggle)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)

        // Set the adapter for the list view
        //final TypedArray typedArray = getResources().obtainTypedArray(R.array.menu_icons);
        //mDrawerList.setAdapter(new ArrayAdapter<String>(
        //        this,
        //        R.layout.drawer_items,
        //        mMenuItems
        //) {
        //@Override
        //public View getView(int position, View convertView, ViewGroup parent) {
        //View v = super.getView(position, convertView, parent);
        //int resourceId = typedArray.getResourceId(position, 0);
        //Drawable drawable = getResources().getDrawable(resourceId);
        //((TextView) v).setCompoundDrawablesWithIntrinsicBounds(drawable, null, null, null);
        //return v;
        //}
        //});

        // Set the list's click listener
        mDrawerList.onItemClickListener = DrawerItemClickListener()
    }

    /**
     * Swaps views in the main content view
     */
    private fun selectItem(position: Int) {
        // Hide all views
        findViewById<View>(R.id.dashboardHome).visibility = View.INVISIBLE
        findViewById<View>(R.id.dashboardClasses).visibility = View.INVISIBLE
        findViewById<View>(R.id.dashboardAssignments).visibility = View.INVISIBLE
        findViewById<View>(R.id.dashboardPayments).visibility = View.INVISIBLE
        findViewById<View>(R.id.dashboardHostel).visibility = View.INVISIBLE
        findViewById<View>(R.id.dashboardSettings).visibility = View.INVISIBLE
        findViewById<View>(R.id.dashboardHelp).visibility = View.INVISIBLE
        when (position) {
            0 -> {
                findViewById<View>(R.id.dashboardHome).visibility = View.VISIBLE
                title = "ActivityDashboard"
            }
            1 -> findViewById<View>(R.id.dashboardClasses).visibility = View.VISIBLE
            2 -> findViewById<View>(R.id.dashboardAssignments).visibility = View.VISIBLE
            3 -> findViewById<View>(R.id.dashboardPayments).visibility = View.VISIBLE
            4 -> findViewById<View>(R.id.dashboardHostel).visibility = View.VISIBLE
            5 -> findViewById<View>(R.id.dashboardSettings).visibility = View.VISIBLE
            6 -> findViewById<View>(R.id.dashboardHelp).visibility = View.VISIBLE
        }

        // Highlight the selected item, update the title, and close the drawer
        mDrawerList.setItemChecked(position, true)
        mDrawerLayout.closeDrawer(mDrawerMain)
    }

    override fun setTitle(title: CharSequence) {
        mDrawerTitle = title.toString()
        mTitle = mDrawerTitle
        supportActionBar?.title = title
    }

    private inner class DrawerItemClickListener : OnItemClickListener {
        override fun onItemClick(parent: AdapterView<*>?, view: View, position: Int, id: Long) {
            selectItem(position)
        }
    }
}